xds/src/test/java/io/grpc/xds/ControlPlaneRule.java

```diff
@@ -118,14 +126,16 @@ public Server getServer() {
 
   @Override protected void finished(Description description) {
     if (server != null) {
-      server.shutdownNow();
+      server.shutdown();
```

Comment:
If you're going to swap to `shutdown()` here, then add a `shutdownNow()` in the block where `awaitTermination()` returns false.

Reply:
I went back to shutdownNow() and that alone brought the test down to ~2s.